dotCMS/src/main/java/com/dotcms/vanityurl/model/CachedVanityUrl.java

```diff
@@ -45,113 +53,40 @@ public CachedVanityUrl(final VanityUrl vanityUrl) {
         this.order    = vanityUrl.getOrder();
     }
 
-    /**
-     * Generates a CachedVanityUrl from another given CachedVanityUrl
-     *
-     * @param fromCachedVanityUrl VanityURL to copy
-     * @param url url to override in the created copy
-     */
-    public CachedVanityUrl(CachedVanityUrl fromCachedVanityUrl, String url) {
-
-        this.vanityUrlId = fromCachedVanityUrl.getVanityUrlId();
-        //if the VanityUrl URI is not a valid regex
-        final String regex = normalize(url, CACHE_404_VANITY_URL.equals(this.vanityUrlId));
+    public CachedVanityUrl(String vanityUrlId,String url, long languageId, String siteId,String forwardTo,int response) {
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Parameter 'vanityUrlId' is not assigned and could be declared final](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=5434315)